File: 0507-perfect-number/0507-perfect-number.java
class Solution {
    public boolean checkPerfectNumber(int num) {
        ArrayList<Integer> lst = new ArrayList<>();
        int sum = 0;
        for(int i=1; i<num; i++){
            if(num % i == 0){
                lst.add(i);
            }
        }
        for(int i=0; i<lst.size(); i++){
            sum += lst.get(i);
        }
        return sum == num;
        
    }
}

File: 0507-perfect-number/0507-perfect-number.cpp
class Solution {
public:
    bool checkPerfectNumber(int n) {
        if (n == 1) return false;

        int sum = 0;

        for (int i = 1; i * i <= n; i++) {
            if (n % i == 0) {
                sum += i;

                int other = n / i;
                if (other != i && other != n) {
                    sum += other;
                }
            }
        }

        return sum == n;
    }
};